proxy/http/HttpBodyFactory.cc

```diff
@@ -411,7 +411,7 @@ HttpBodyFactory::fabricate(StrList *acpt_language_list, StrList *acpt_charset_li
     set = determine_set_by_language(acpt_language_list, acpt_charset_list);
   } else if (enable_customizations == 3) {
     set = determine_set_by_host(context);
-  } else if (is_response_body_precluded(context->http_return_code, context->method)) {
+  } else if (is_response_body_precluded(context->http_return_code)) {
```

Comment:
IMHO, I don't think we need to add is_response_body_precluded there. For example if http_return_code is HTTP_STATUS_NOT_MODIFIED. It also return resulting_buffer_length with 0.

Then ats also close the connection directly.